usage/usage.py

```diff
@@ -11,459 +11,20 @@
-        edges = value[:1] + sum(gap, []) + value[-1:]
-        start_end[key] = list(zip(edges[::2], edges[1::2]))
-
-    return start_end
-
-
-def get_nodehours(data):
-    """Takes the accounting data from SLURM and outputs the node-hours used for
-    all of the slurm jobs requested.
-
-    Parameters
-    ----------
-    data : `list` of `dict`
-        accounting data from slurm.
-
-    Returns
-    -------
-    node_hours : `float`
-        the total node-hours spent on all of the slurm jobs passed into
-        usage.py.
-    """
-    # Find duration of each job
-    duration = [(rec['end'] - rec['start']).total_seconds() for rec in data]
-    nodes = [int(rec['nnodes']) for rec in data]
-
-    # Collect elapsed time data
-    node_hours = sum(dur*node for dur, node in zip(duration, nodes))
-
-    return round(node_hours/3600.0, 2)
-
-
-def get_codehours(data):
-    """Finds the elapsed node-hours for each code (ie, how much time it took to
-    complete all coadd jobs multiplied by the nodes used, etc.).
-
-    Parameters
-    ----------
-    data : `list` of `dict`
-        accounting data from slurm.
-
-    Returns
-    -------
-    code_nodehours : `dict` of `str` keys and `float` values
-        dictionary containing the code names and their associated node-hours
-        on SLURM in hours*nodes
-    """
-    # Find duration of each job & nodes used
-    duration = [(rec['end'] - rec['start']).total_seconds() for rec in data]
-    nodes = [int(rec['nnodes']) for rec in data]
-    job_type = [rec['jobname'] for rec in data]
-
-    # Return elapsed_times
-    code_nodehours = dict.fromkeys(job_type, 0.0)
-    for dur, name, node in zip(duration, job_type, nodes):
-        code_nodehours[name] += dur*node
-    return {key: round(val/3600.0, 2) for key, val in code_nodehours.items()}
-
-
-def convert_names(data, mapping):
-    """Convert names from JobNames to their code names.
-
-    Parameters
-    ----------
-    data : `list` of `dict`
-        accounting data from slurm.
-    mapping : `dict`
-        dictionary mapping jobName prefixes to code names
-    """
-    # Extract mapping keys for matching
-    cores = mapping.keys()
-    for datum in data:
-        name = datum['jobname']
-
-        # See if name matches any keys
-        matches = [re.match(core, name) for core in cores]
-        matches = [match[0] for match in matches if match]
-
-        # Find new name or throw error if too many matches
-        if len(matches) == 0:
-            code = 'unknown'
-        elif len(matches) == 1:
-            code = mapping[matches[0]]
-        else:
-            msg = 'ERROR: Ambiguous mapping: ' \
-                  'following keys "%s" can be mapped to "%s".' % \
-                  (', '.join(match for match in matches), name)
-            raise RuntimeError(msg)
-
-        # Assign new code name
-        datum['jobname'] = code
+import data as d
+import output as out
```

Comment:
I do not really see a need for using shortcuts; `data` is much better name to use globally in the module comparing to enigmatic `d`; similiary  with `output`.